### PR TITLE
User Management - Front-end

### PR DESCRIPTION
**Pull request by sanjaytkbabu**
# Description
[PADS-176](https://apps.nrs.gov.bc.ca/int/jira/browse/PADS-176)
New view & modals added for user management
New permission added for granularity

## Types of changes

New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have checked that unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by slhurley** [comment]
Looks good for initial front-end work.  Only comment is that the Revoke modal for Admins needs to say a different message than it does for Supervisors.  ("The user will now lose all access...." message)

**Comment by sanjaytkbabu** [comment]
> Looks good for initial front-end work. Only comment is that the Revoke modal for Admins needs to say a different message than it does for Supervisors. ("The user will now lose all access...." message)

done!

File: frontend/src/components/user/UserManageModal.vue [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+      />
+      <Button
+        class="p-button-outlined mr-2"
+        label="Cancel"
+        icon="pi pi-times"
+        @click="
+          () => {
+            visible = false;
+          }
+        "
```
Comment (wilwong89) [inline frontend/src/components/user/UserManageModal.vue:52]: Can simplify this to `@click='visible = false'` unless there needs to be multiple statements to execute.

File: frontend/src/views/user/UserManagementView.vue [hunk 2]
```diff
@@ -0,0 +1,178 @@
+<script setup lang="ts">
+import {
+  Button,
+  FilterMatchMode,
+  IconField,
+  InputIcon,
+  InputText,
+  useConfirm,
+  TabPanel,
+  TabView
```
Comment (wilwong89) [inline frontend/src/views/user/UserManagementView.vue:10]: `useConfirm` should go after `TabView`

File: frontend/src/components/user/UserCreateModal.vue [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+    <Dropdown
+      class="col-12"
+      name="assignRole"
+      label="Assign role"
+      :options="ROLES"
+    />
+    <div class="flex-auto">
+      <Button
+        class="mr-2"
+        label="Request Approval"
```
Comment (wilwong89) [inline frontend/src/components/user/UserCreateModal.vue:85]: Figma has lowercase A for the second word.

File: frontend/src/components/user/UserCreateModal.vue [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+      />
+      <Button
+        class="p-button-outlined mr-2"
+        label="Cancel"
+        icon="pi pi-times"
+        @click="
+          () => {
+            visible = false;
+          }
+        "
```
Comment (wilwong89) [inline frontend/src/components/user/UserCreateModal.vue:103]: Can simplify this to @click='visible = false' unless there needs to be multiple statements to execute like above in line 89.

**Comment by sanjaytkbabu** [comment]
comments addressed!

File: frontend/src/components/user/UserTable.vue [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+    >
+      <template #body="{ data }">
+        <Button
+          class="p-button-lg p-button-text p-0 pr-3"
+          aria-label="Manage user"
+          :disabled="revocation || data.status !== USER_STATUS.APPROVED"
+          @click="
+            selection = data;
+            emit('userTable:manage', data);
+          "
```
Comment (wilwong89) [inline frontend/src/components/user/UserTable.vue:89]: Optional: use `()=>{ selection = data; emit('userTable:manage', data);}`
Mainly just a consistency thing with other multi-statement handling elsewhere in the file.

**Review by wilwong89 — APPROVED** [review]

File: frontend/src/components/user/UserTable.vue (hunk 5 above)
Comment (sanjaytkbabu) [reply frontend/src/components/user/UserTable.vue]: done!

**Review by kyle1morel — APPROVED** [review]

**Pull request merged**